server/src/main/java/org/elasticsearch/action/admin/cluster/node/capabilities/TransportNodesCapabilitiesAction.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.action.admin.cluster.node.capabilities;
+
+import org.elasticsearch.action.ActionType;
+import org.elasticsearch.action.FailedNodeException;
+import org.elasticsearch.action.support.ActionFilters;
+import org.elasticsearch.action.support.nodes.TransportNodesAction;
+import org.elasticsearch.cluster.node.DiscoveryNode;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.core.RestApiVersion;
+import org.elasticsearch.rest.RestController;
+import org.elasticsearch.rest.RestRequest;
+import org.elasticsearch.tasks.Task;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.elasticsearch.transport.TransportRequest;
+import org.elasticsearch.transport.TransportService;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Set;
+
+public class TransportNodesCapabilitiesAction extends TransportNodesAction<
+    NodesCapabilitiesRequest,
+    NodesCapabilitiesResponse,
+    TransportNodesCapabilitiesAction.NodeCapabilitiesRequest,
+    NodeCapability> {
+
+    public static final ActionType<NodesCapabilitiesResponse> TYPE = new ActionType<>("cluster:monitor/nodes/capabilities");
+
+    private final RestController restController;
+
+    @Inject
+    public TransportNodesCapabilitiesAction(
+        ThreadPool threadPool,
+        ClusterService clusterService,
+        TransportService transportService,
+        ActionFilters actionFilters,
+        RestController restController
+    ) {
+        super(
+            TYPE.name(),
+            clusterService,
+            transportService,
+            actionFilters,
+            NodeCapabilitiesRequest::new,
+            threadPool.executor(ThreadPool.Names.MANAGEMENT)
+        );
+        this.restController = restController;
+    }
+
+    @Override
+    protected NodesCapabilitiesResponse newResponse(
+        NodesCapabilitiesRequest request,
+        List<NodeCapability> responses,
```

Comment:
More for my own curiosity, I don't think it's necessary for this operation.
Is there a way / example how to short circuit and respond immediately as soon as 1 node returns unsupported?

Reply:
I don't think that's functionality supported by the base classes, it would be a significant re-implementation of `CancellableFanOut` to do so.

Reply:
ok, the pattern is probably just too specific